src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs

```diff
@@ -103,7 +103,6 @@ public async Task<ILocalGitRepo> PrepareCloneAsync(
         }
 
         var repo = await PrepareCloneInternalAsync(mapping.Name, remoteUris, [checkoutRef], checkoutRef, resetToRemote, cancellationToken);
-        await repo.CheckoutAsync(checkoutRef);
```

Comment:
this isn't needed anymore because the method we're calling checks out the ref for us